lib-injection/sitecustomize.py

```diff
@@ -21,7 +21,8 @@
 
 
 def parse_version(version: str) -> Tuple:
-    constraint_idx = re.search(r"\d", version).start()
+    v = re.search(r"\d", version)
```

Comment:
This was required to fix a linting error. `re.search(r"\d", version)` can return `None`.  We need to do a none check before calling `.start()`.

Yesterday I was experimenting with expanding the scope of this PR to address @romainkomorndatadog's concerns. It would be nice if one hatch script encapsulated all the checks run in pre-check. This will allow us to easily run pre-check locally. I will revert this change and avoid updating `gen_circleci_config.py`. I'll limit the scope of this PR to adding `lint:checks` to `hatch.toml`. This should be a good first step.